### PR TITLE
Issue #2233759: Make it possible to reset all data for an agent from the...

### PR DESCRIPTION
**Pull request by katbailey**
... campaign page

### CODE REVIEW

File: acquia_lift.module [hunk 1]
```diff
@@ -967,6 +967,58 @@ function acquia_lift_navbar() {
...
+      '#prefix' => '<div id="personalize-acquia-lift-reset">',
+      '#suffix' => '</div>',
+      '#type' => 'submit',
+      '#name' => 'reset_submit',
+      '#value' => t('Reset data'),
+      '#attributes' => array(
+        'class' => array('action-item-primary-active'),
+      ),
+      '#ajax' => array(
+        'callback' => 'personalize_acquiq_lift_ajax_callback',
```
Comment (eshtadc) [inline acquia_lift.module:987]: acquiq (although you are consistent - the callback is acquiq too)

File: includes/acquia_lift.classes.inc [hunk 2]
```diff
@@ -415,6 +415,25 @@ class AcquiaLiftAPI implements PersonalizeLoggerAwareInterface {
...
+    $vars = array('agent' => $agent_name);
+    $success_msg = 'The data for Acquia Lift campaign {agent} was reset';
+    $fail_msg = 'Could not reset data for Acquia Lift campaign {agent}';
+    if ($response->code == 200) {
+      $this->logger()->log(PersonalizeLogLevel::INFO, $success_msg, $vars);
+    }
+    else {
+      $this->logger()->log(PersonalizeLogLevel::ERROR, $fail_msg, $vars);
+    }
+  }
```
Comment (eshtadc) [inline includes/acquia_lift.classes.inc:434]: This is probably an issue on its own instead, but here we know if there is an error or not, but we don't return an error code.  Instead in the calling function we check for an exception being thrown, but couldn't the error just be that we get an invalid response code?  Seems like the user could get a message that the data was reset when it wasn't.

File: acquia_lift.module (hunk 1 above)
Comment (katbailey) [inline acquia_lift.module:987]: Copy/paste has its benefits: my typos still _work_ :) Nice catch

File: includes/acquia_lift.classes.inc (hunk 2 above)
Comment (katbailey) [inline includes/acquia_lift.classes.inc:434]: Ooh, yeah that is dumb. Most of the PUT/DELETE requests are made asynchronously by the queue runner so we don't bother throwing an exception. It does make sense to throw it here though. The queue processor catches any exception thrown so maybe I should make them all throw an exception on failure.

**Pull request closed without merging**